node-src/tasks/build.ts

```diff
@@ -69,14 +69,23 @@ export const buildStorybook = async (ctx: Context) => {
     logFile.on('error', reject);
   });
 
+  const { experimental_abortSignal: abortSignal } = ctx.options;
+
   try {
     const { command } = ctx.spawnParams;
     ctx.log.debug('Using spawnParams:', JSON.stringify(ctx.spawnParams, null, 2));
-    await Promise.race([
-      execa.command(command, { stdio: [null, logFile, logFile] }),
-      timeoutAfter(ctx.env.STORYBOOK_BUILD_TIMEOUT),
-    ]);
+
+    let subprocess;
+    if (abortSignal) {
+      abortSignal.onabort = () => {
+        subprocess?.kill('SIGTERM', { forceKillAfterTimeout: 2000 });
+      };
+    }
```

Comment:
execa has [supported an abort signal since `6.1.0`](https://github.com/sindresorhus/execa/releases/tag/v6.1.0). Maybe we should upgrade?

Reply:
Oh nice! I'll look into that.